python/tk_framework_desktopserver/shotgun/scripts/cache_commands.py

```diff
@@ -257,7 +255,25 @@ def cache(
     with open(arg_data_file, "rb") as fh:
         arg_data = cPickle.load(fh)
 
-    sys.path = sys.path + arg_data["sys_path"]
+    # We do this in two steps. The first is to prepend to sys.path
+    # before immediately importing sgtk. After that, we replace the
+    # prepend with an append. This allows us to guarantee that we
+    # have sgtk imported from the right place, followed by setting
+    # up sys.path the way that is safest for other imports.
+    original_sys_path = copy.copy(sys.path)
+
+    try:
+        # The RPC api that spawned us has ensured that the first path
+        # in the sys.path list that it provided us is the root path that
+        # will get us the tk-core that we want. We can't prepend any more
+        # than that, because we might be running in a different version
+        # of Python in this process than SG Desktop is. We don't want to
+        # force Python 2.7 standard library modules to be imported if
+        # we're in Python 2.6 here, as an example.
+        sys.path = [arg_data["sys_path"][0]] + sys.path
+        import sgtk
+    finally:
+        sys.path = original_sys_path + arg_data["sys_path"]
```

Comment:
how self contained is this script? Why exactly do we even need to append `arg_data["sys_path"]` at all? The comments are great, and indeed very verbose, yet i think it would still be useful with a concrete example of why we need `arg_data["sys_path"]` to be appended to this process' `sys.path`

Reply:
I think you're almost certainly correct that it's overkill. I'm shifting to only passing over the tk-core root path so that sgtk can be imported, and not doing anything else with sys.path.